### DISCUSSION TITLE
Ensure switch is ON between 9pm and midnight

### DISCUSSION THREAD

**Discussion by AdrianGin** [arendst/Tasmota#20985] [Q&A] [answered]
I wish to ensure that my switch is on during 9pm and midnight each day.

However due to ripple control, the switch might only turn on at 11pm. This means that a conventional timer might miss the "ON" event at 9pm, due to it not being powered.

How can I setup a rule such that on powerup, check the time, and if it's between 9pm and midnight then set power to ON, otherwise turn off.

**Comment by barbudor** [comment, marked as answer]
Simplest one : if the ripple control always ends at 11:00pm, just configure a 2nd ON timer event at 11:01pm

Other option, considering that your OFF is at midnight, generate an event at boot
Something like
```
Rule1 
   ON system#boot DO event boottime=%time% ENDON
   ON event#boottime>=1380 DO Power ON ENDON
```

Should work (not tested)

**Reply by sfromis** [reply]
You mentioned two different times..... For the trigger time being 21:00 instead of 23:00, use the number 1260 instead of 1380, it is simply minutes past midnight. 

If you want to also power off if boot time was before 21:00, you can add a rule like:
`ON event#boottime<1260 DO Power ON ENDON`

**Reply by AdrianGin** [reply]
Thanks! This rule works great!
